toolRepository-remote/src/main/scala/com/codacy/toolRepository/remote/ToolRepositoryRemote.scala

```diff
@@ -88,14 +90,27 @@ class ToolRepositoryRemote(toolsClient: ToolsClient,
         case e: Exception => Left(AnalyserError.FailedToListPatterns(tool.uuid, e.getMessage))
       }
 
-    val result = Await.result(patternsF, 1 minute)
-    val patternSpecDataStorage = patternStorage(tool.uuid)
-    result match {
-      case Right(patterns) =>
-        patternSpecDataStorage.save(patterns)
-        Right(patterns)
-      case Left(err) =>
-        patternSpecDataStorage.get().toRight(err)
+    Await.result(patternsF, 1 minute)
+  }
+
+  override def listPatterns(tool: ToolSpec): Either[AnalyserError, Seq[PatternSpec]] = {
+    val toolPatternsStorageFilename = s"${tool.uuid}-${tool.version}"
+    val toolPatternsStorageInstance = patternStorage(toolPatternsStorageFilename)
+
+    toolPatternsStorageInstance.get() match {
+      case Some(patternsFromStorage) =>
+        logger.info(s"Using patterns from cache for ${tool.name} version ${tool.version}")
+        Right(patternsFromStorage)
+      case None =>
+        patternsFromApi(tool) match {
```

Comment:
Maybe this could be extracted to a method. Just to avoid the nested pattern matching. What do you think?

Reply:
Yep, it gets cleaner 👍 
Extracted it to a different method.